### PR TITLE
Some changes recommended by CLion's analyzer

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: lib/fizzy/types.hpp [hunk 1]
```diff
@@ -294,7 +294,7 @@ struct ConstantExpression
     Kind kind = Kind::Constant;
     union
     {
-        Value constant = 0;
+        Value constant{0};
```
Comment (chfast) [inline lib/fizzy/types.hpp:297]:
```suggestion
        Value constant{};
```

File: lib/fizzy/instructions.cpp [hunk 2]
```diff
@@ -448,12 +448,12 @@ constexpr uint8_t instruction_max_align_table[256] = {
 
 const InstructionType* get_instruction_type_table() noexcept
 {
-    return instruction_type_table;
+    return static_cast<const InstructionType*>(instruction_type_table);
```
Comment (chfast) [inline lib/fizzy/instructions.cpp:451]: This does not make much sense.

File: lib/fizzy/trunc_boundaries.hpp [hunk 3]
```diff
@@ -51,24 +50,22 @@ template <>
...
+    static constexpr float lower = -1.0f;
+    static constexpr float upper = 18446744073709551616.0f;
 };
 
 template <>
 struct trunc_boundaries<double, int64_t>
 {
     /// The first representable value lower than theoretical -9223372036854775809.
     static constexpr double lower = -9223372036854777856.0;
-
```
Comment (chfast) [inline lib/fizzy/trunc_boundaries.hpp:71]: What is the problem here?

Comment (axic) [reply lib/fizzy/trunc_boundaries.hpp:71]: Here, nothing, just some cases had an empty line other don't, made that consistent.

Comment (chfast) [reply lib/fizzy/trunc_boundaries.hpp:71]: To separate elements with comments.

Comment (axic) [reply lib/fizzy/trunc_boundaries.hpp:71]: What do you mean? Doxygen?

File: lib/fizzy/value.hpp [hunk 4]
```diff
@@ -71,13 +71,13 @@ constexpr int32_t Value::as<int32_t>() const noexcept
 }
 
 template <>
-constexpr float Value::as() const noexcept
+constexpr float Value::as<float>() const noexcept
```
Comment (chfast) [inline lib/fizzy/value.hpp:74]: I'm surprised it worked before.

File: lib/fizzy/execute.hpp [hunk 5]
```diff
@@ -20,7 +20,7 @@ struct ExecutionResult
 {
     const bool trapped = false;
     const bool has_value = false;
-    const Value value = 0;
+    const Value value{};
```
Comment (gumb0) [inline lib/fizzy/execute.hpp:23]: @chfast After this default-initialization which member of `Value` is it allowed to access without UB?

File: lib/fizzy/trunc_boundaries.hpp (hunk 3 above)
Comment (gumb0) [reply lib/fizzy/trunc_boundaries.hpp:71]: I understood that it's to show that the comment refers to the next line only, and doesn't to the line after.

File: lib/fizzy/execute.hpp (hunk 5 above)
Comment (chfast) [reply lib/fizzy/execute.hpp:23]: > If T is a union type, the first non-static named data member is zero-initialized and all padding is initialized to zero bits. 

Maybe that is not the best idea in the end.

File: lib/fizzy/value.hpp (hunk 4 above)
Comment (axic) [reply lib/fizzy/value.hpp:74]: There's a lot of surprising things

File: lib/fizzy/execute.hpp (hunk 5 above)
Comment (axic) [reply lib/fizzy/execute.hpp:23]: So what should be done here?

Comment (chfast) [reply lib/fizzy/execute.hpp:23]: I did some experiments and this is fine.

File: lib/fizzy/trunc_boundaries.hpp (hunk 3 above)
Comment (chfast) [reply lib/fizzy/trunc_boundaries.hpp:71]: Yes, but the new way is fine too.

File: lib/fizzy/execute.hpp (hunk 5 above)
Comment (gumb0) [reply lib/fizzy/execute.hpp:23]: At some point documentation for `ExecutionResult` shoud be added, explaining how it is allowed to access `value` (only when  `!trapped & has_value` and only as the correct type)

**Pull request merged**